src/standalone/Soil/energy_hydrology.jl

```diff
@@ -2,54 +2,48 @@
     EnergyHydrologyParameters{FT <: AbstractFloat}
 
 A parameter structure for the integrated soil water and energy
- equation system. In this simplest
-form, we assume the conductivity and volumetric heat capacity
-of the soil are constant.
+ equation system. 
 $(DocStringExtensions.FIELDS)
 """
-struct EnergyHydrologyParameters{
-    FT <: AbstractFloat,
-    C <: AbstractSoilHydrologyClosure,
-    PSE,
-}
+struct EnergyHydrologyParameters{FT <: AbstractFloat, F, C, PSE}
```

Comment:
for RichardsParameters we specify `F <: Union{AbstractFloat, ClimaCore.Fields.Field}` - it might be nice to do that here too